### ISSUE TITLE
quantization convert should warn the user if calibration has not happened

### ISSUE THREAD

**Issue by vkuzo** [pytorch/pytorch#88609]
### 🐛 Describe the bug

It's an easy mistake to pass the wrong model into `prepare_fx`, such as

```
import copy
import torch
import torch.ao.quantize_fx as quantize_fx

m = M(...)
mp = prepare_fx(copy.deepcopy(m), ...)
# wrong model used for calibration
calibrate(m, data_loader)
mq = convert_fx(mp)
evaluate(mq, data_loader)
```

Currently there is no way for user to realize the mistake until after they evaluated the accuracy, which can take awhile depending on model/dataset.  If the user is doing static quantization, we should warn the user during convert if any activation observers have not seen any data.

### Versions

master

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @jgong5 @Xia-Weiwen @leslie-fang-intel

**Comment by jon-chuang** [comment]
> we should warn the user during convert if any activation observers have not seen any data.

I think this is unnecessary. We should do two things:
1. assert the model passed to convert_fx `is_observed_module`
2. wrap the`forward` function of `ObservedGraphModule` to indicate a property `recorded_forward_pass` which is set to true if forward is called on the model

**Comment by vkuzo** [comment]
cc @jerryzh168 , I think there are plans to remove the `ObservedGraphModule` types in the near future.

Overall I'm very flexible on the "how", the issue is more to note down that this is an easy mistake to make, and it would be good to add a warning.

**Comment by jerryzh168** [comment]
I remember we already have warnings when we call calculate parameters on observer when it haven't go through calibration step: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/observer.py#L1204

**Comment by andrewor14** [comment]
> I remember we already have warnings when we call calculate parameters on observer when it haven't go through calibration step: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/observer.py#L1204

@jerryzh168 Do you mind updating the link to point to the correct line? Looks like the lines have shifted since your comment.

**Comment by HDCharles** [comment]
it looks like it does
